src/fetch_foursquare_info.py

```diff
@@ -32,98 +37,99 @@ def load_cnpjs(subquota_description):
 
-def load_companies():
+def load_companies_dataset(cnpjs):
+    """Return a DataFrame of companies from the given list of cnpjs"""
     u_cols = ['cnpj', 'trade_name', 'zip_code', 'latitude', 'longitude']
     all_companies = pd.read_csv(COMPANIES_DATASET_PATH,
                                 low_memory=False,
                                 usecols=u_cols,
                                 dtype={'trade_name': np.str})
-    all_companies['clean_cnpj'] = all_companies['cnpj'].map(cleanup_cnpj)
-    relevant_cnpjs = load_cnpjs('Congressperson meal')
-    return all_companies[all_companies['clean_cnpj'].isin(relevant_cnpjs)]
+    all_companies['clean_cnpj'] = all_companies['cnpj'].map(only_numbers)
+    return all_companies[all_companies['clean_cnpj'].isin(cnpjs)]
 
-def cleanup_cnpj(cnpj):
-    return re.sub("\D", "", cnpj)
+def only_numbers(string):
+    """Return a string w only the numbers from the given string"""
+    return re.sub("\D", "", string)
 
-def remaining_companies(companies):
-    fetched_companies = load_foursquare_companies_dataset()
+def remaining_companies(companies, fetched_companies):
+    """Return the first DF but without matching CNPJs from the second DF"""
     return companies[~companies['cnpj'].isin(fetched_companies['cnpj'])]
 
 def load_foursquare_companies_dataset():
+    """Return a DF with the data already collected"""
     if os.path.exists(FOURSQUARE_DATASET_PATH):
-        return pd.read_json(FOURSQUARE_DATASET_PATH)
+        return pd.read_csv(FOURSQUARE_DATASET_PATH)
     else:
         return pd.DataFrame(columns=['cnpj'])
 
 def get_venue(company):
+    """Return a matching venue from Foursquare for the given company Series"""
     venue = search(company)
     if venue:
-        return fetch_venue_by_id(venue['id'])
+        return fetch_venue(venue['id'])
 
-def search(company, timeout=5):
+def search(company):
+    """Search Foursquare's API for a match for a given company Series"""
     params = dict(DEFAULT_PARAMS)
-    params.update({
-        'query': company['trade_name'],
-        'll': '%s,%s' % (company['latitude'], company['longitude']),
-        'zip_code': company['zip_code'],
-        'intent': 'match'
-    })
+    params.update({ 'query': company['trade_name'],
+                    'll': '%s,%s' % (company['latitude'], company['longitude']),
+                    'zip_code': company['zip_code'],
+                    'intent': 'match' })
     url = 'https://api.foursquare.com/v2/venues/search'
     response = requests.get(url, params=params)
-
-    return parse_search_results(response)
+    if 200 >= response.status_code < 300:
+        return parse_search_results(response)
 
 def parse_search_results(response):
+    """Return the first venue from the given search response"""
     json = response.json()
-    if json['meta']['code'] == 200:
-        venues = json['response']['venues']
-        if venues:
-            return venues[0]
+    venues = json['response']['venues']
```

Comment:
This might be a bit of 💅 but…

If you get a JSON a bit different than expected `json['response']['venues']` will crash with a `KeyError`. You could use [Dictionary's `get` method](https://docs.python.org/3/library/stdtypes.html?highlight=get#dict.get) to overcome that:

```python
json_response = response.json()
venues = json_response.get('response', {}).get('venues')
```

(In the first `get` I'm setting the `default` to an empty dictionary (`{}`) so if the key `response` is missing the returned object  still have the `get` method (if the first `get` returns `None` the second `get` fails because `None` não tem o método `get`).